src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -55,36 +55,70 @@ internal static IEnumerable<CompletionResult> GetMatchingResults(
+        /// If wrapping quotes are detected and match, both are removed; otherwise, only the front quote is removed.
+        /// The string is updated in-place, and only matching front-and-back quotes are stripped.
+        /// If no quotes are detected or the input is empty, the original string remains unchanged.
+        /// </remarks>
         internal static string HandleDoubleAndSingleQuote(ref string wordToComplete)
         {
-            string quote = string.Empty;
+            if (string.IsNullOrEmpty(wordToComplete))
+            {
+                return string.Empty;
+            }
+
+            char frontQuote = wordToComplete[0];
+            bool hasFrontSingleQuote = frontQuote.IsSingleQuote();
+            bool hasFrontDoubleQuote = frontQuote.IsDoubleQuote();
 
-            if (!string.IsNullOrEmpty(wordToComplete) && (wordToComplete[0].IsSingleQuote() || wordToComplete[0].IsDoubleQuote()))
+            if (!hasFrontSingleQuote && !hasFrontDoubleQuote)
             {
-                char frontQuote = wordToComplete[0];
-                int length = wordToComplete.Length;
+                return string.Empty;
+            }
 
-                if (length == 1)
-                {
-                    wordToComplete = string.Empty;
-                    quote = frontQuote.IsSingleQuote() ? "'" : "\"";
-                }
-                else if (length > 1)
-                {
-                    if ((wordToComplete[length - 1].IsDoubleQuote() && frontQuote.IsDoubleQuote()) || (wordToComplete[length - 1].IsSingleQuote() && frontQuote.IsSingleQuote()))
-                    {
-                        wordToComplete = wordToComplete.Substring(1, length - 2);
-                        quote = frontQuote.IsSingleQuote() ? "'" : "\"";
-                    }
-                    else if (!wordToComplete[length - 1].IsDoubleQuote() && !wordToComplete[length - 1].IsSingleQuote())
-                    {
-                        wordToComplete = wordToComplete.Substring(1);
-                        quote = frontQuote.IsSingleQuote() ? "'" : "\"";
-                    }
-                }
+            string quoteInUse = hasFrontSingleQuote ? "'" : "\"";
+
+            int length = wordToComplete.Length;
+            if (length == 1)
+            {
+                wordToComplete = string.Empty;
+                return quoteInUse;
+            }
+
+            char backQuote = wordToComplete[length - 1];
+            bool hasBackSingleQuote = backQuote.IsSingleQuote();
+            bool hasBackDoubleQuote = backQuote.IsDoubleQuote();
+
+            bool hasMatchingFrontAndBackQuote =
+                (hasFrontSingleQuote && hasBackSingleQuote) || (hasFrontDoubleQuote && hasBackDoubleQuote);
+
+            if (hasMatchingFrontAndBackQuote)
+            {
+                wordToComplete = wordToComplete.Substring(1, length - 2);
+                return quoteInUse;
+            }
+
+            bool hasValidFrontQuoteAndNoMatchingBackQuote = 
```

Comment:
Maybe `hasFrontQuoteAndNoBackQuote`?

Reply:
@iSazonov Thanks. Like this much better. Fixed in d0bb4f30ead67dd9c4f8f23b33d4eb5a6fc7d67f